### PR TITLE
Feat/authentication

### PR DESCRIPTION
**Pull request by thaisribon**
# Description

- Authentication

# Necessary changes

- check google client id env vars

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

Put here all the things need to test this PR and verify your changes, also list any relevant details for tests (eg: have a wallet)

- Test donation with old user and new user and with all methods
- Test create account with all methods
- Test link expired
- Test logout
- Test extra ticket, check if it is possible get more tickets with that integration
- Test delete account

### CODE REVIEW

File: src/components/moleculars/buttons/AppleLogin/index.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  const { t } = useTranslation("translation", {
+    keyPrefix: "components.moleculars.buttons.AppleLogin",
+  });
+
+  const { signInWithApple } = useAuthentication();
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const handleApple = async (response: any) => {
+    logEvent("authAppleBtn_click", {
+      from: "donation_flow",
```
Comment (letbueno) [inline src/components/moleculars/buttons/AppleLogin/index.tsx:26]: adjust from on event to receive as param

File: src/components/moleculars/buttons/GoogleLogin/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if (error.message.includes("Email does not match")) {
+          setModalVisible(true);
+        }
+      }
+    },
+  });
+
+  function handleGoogle() {
+    logEvent("authGoogleBtn_click", {
+      from: "donation_flow",
```
Comment (letbueno) [inline src/components/moleculars/buttons/GoogleLogin/index.tsx:39]: here too

File: src/components/moleculars/buttons/MagicLinkLogin/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  onContinue: () => void;
+};
+function MagicLinkLogin({ onContinue }: Props): JSX.Element {
+  const { t } = useTranslation("translation", {
+    keyPrefix: "components.moleculars.buttons.MagicLinkLogin",
+  });
+
+  const handleMagicLink = () => {
+    logEvent("authEmailBtn_click", {
+      from: "donation_flow",
```
Comment (letbueno) [inline src/components/moleculars/buttons/MagicLinkLogin/index.tsx:17]: here too

File: src/hooks/modalHooks/useAnimationReceiveTicketModal/index.tsx [hunk 4]
```diff
@@ -20,17 +21,20 @@ export function useAnimationReceiveTicketModal(initialState?: boolean) {
...
+  const { profile } = userProfile();
 
   const { show, hide } = useModal({
     type: MODAL_TYPES.MODAL_ANIMATION,
     props: {
       text: t("receiveTicketAnimationModalTitle"),
       iconOrigin: SupportersIcon,
       textOrigin: t("receiveTicketAnimationModalOrigin"),
-      iconDestiny: UserIcon,
+      iconDestiny: profile?.photo ?? UserIcon,
```
Comment (letbueno) [inline src/hooks/modalHooks/useAnimationReceiveTicketModal/index.tsx:33]: i think that name is confuse

File: src/lib/contactSupport/index.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Languages } from "@ribon.io/shared/types";
+import startSupportChat from "services/support";
+import { USER_SUPPORT_LINK } from "utils/constants";
+
+function contactSupport(language: Languages) {
+  if (language === Languages.PT) {
+    window.open(USER_SUPPORT_LINK, "_blank");
```
Comment (letbueno) [inline src/lib/contactSupport/index.ts:7]: i think we should add a event

File: src/pages/donations/auth/SignedInPage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+  const onAnimationEnd = useCallback(() => {
+    if (donationSucceeded) {
+      logEvent("ticketDonated_end", {
+        nonProfitId: nonProfit.id,
+      });
+      navigateTo({
+        pathname: "/ticket-donation-done",
+        state: {
+          flow: "signedIn",
```
Comment (letbueno) [inline src/pages/donations/auth/SignedInPage/index.tsx:57]: i think we should change the signedin in this pull requests

ConfirmDonation?

File: src/hooks/modalHooks/useAnimationReceiveTicketModal/index.tsx (hunk 4 above)
Comment (thaisribon) [reply src/hooks/modalHooks/useAnimationReceiveTicketModal/index.tsx:33]: which name?

**Review by justjuju-me — APPROVED** [review]

**Pull request merged**